### PR TITLE
Loki: Show configured log line limit

### LINKED ISSUE #60295: Loki: Show configured line limit
It would be good to show the currently configured line limit in the query options instead of `auto`.

Image

Also showing it when the options panel is collapsed like we do with the query type might be a way to show that to a user.

NB: we might think of a certain threshold were we display a hint that this line limit might lead to poor performance in the browser.

More info: https://raintank-corp.slack.com/archives/CPXKS6FB5/p1671003302601399

### PR DESCRIPTION
**Pull request by itsgareth**
**What is this feature?**
this feat changes the log line limit placeholder from `auto` to the current line limit (either ds config, or the default) `1000`.
this also always displays the line limit even if query builder options are collapsed.

**Why do we need this feature?**
...

**Who is this feature for?**
...

**Which issue(s) does this PR fix?**:
Fixes #60295

**Special notes for your reviewer**:

from the linked issue: `"we might think of a certain threshold were we display a hint that this line limit might lead to poor performance in the browser."`.

i don't currently have enough experience to know may many logs you can render before getting any performance issues. however if you would like this implemented i'm happy to discuss to find that threshold.

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx [hunk 1]
```diff
@@ -165,7 +165,13 @@ export const LokiQueryEditor = React.memo<LokiQueryEditorProps>((props) => {
             showExplain={explain}
           />
         )}
-        <LokiQueryBuilderOptions query={query} onChange={onChange} onRunQuery={onRunQuery} app={app} />
+        <LokiQueryBuilderOptions
+          query={query}
+          onChange={onChange}
+          onRunQuery={onRunQuery}
+          app={app}
+          dsMaxLines={datasource.maxLines}
```
Comment (svennergr) [inline public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx:173]: What do you think about naming this just `lineLimit` or `maxLines`?

File: public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderOptions.tsx [hunk 2]
```diff
@@ -15,9 +15,10 @@ export interface Props {
   onChange: (update: LokiQuery) => void;
   onRunQuery: () => void;
   app?: CoreApp;
+  dsMaxLines: number;
```
Comment (svennergr) [inline public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderOptions.tsx:18]: Just a nit here, but I think we prefer to list the optional properties in an object last.

File: public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderOptions.tsx [hunk 3]
```diff
@@ -109,8 +115,8 @@ function getCollapsedInfo(query: LokiQuery, queryType: LokiQueryType, showMaxLin
 
   items.push(`Type: ${queryTypeLabel?.label}`);
 
-  if (showMaxLines && query.maxLines) {
-    items.push(`Line limit: ${query.maxLines}`);
+  if (showMaxLines) {
+    items.push(`Line limit: ${query.maxLines || dsMaxLines}`);
```
[the remark this answers is not in the document]
Comment (gwdawson) [reply public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderOptions.tsx]: @svennergr i have fixed the ui rendering issue here, but in terms of functionality what would you like to happen here when the user selects a line limit of `0`?

there are two options we could take
1 - render 0 log lines, thats what the user wanted 🧐
2 - ignore the 0 line limit and use on of the fallbacks (`ds config` or `ds default`)

my preferred is `2`, but i would like a second opinion on this.

**Review by svennergr — APPROVED** [review]
LGTM

**Pull request merged**